src/dbos-executor.ts

```diff
@@ -247,17 +245,12 @@ export class DBOSExecutor {
       // Debug mode doesn't need to initialize the DBs. Everything should appear to be read-only.
       if (!this.debugMode) {
         await this.systemDatabase.init();
-        await this.telemetryCollector.init(this.registeredOperations);
         await this.userDatabase.init(); // Skip user DB init because we're using the proxy.
         await this.recoverPendingWorkflows();
       }
     } catch (err) {
-      if (err instanceof Error) {
-        this.logger.error(`failed to initialize workflow executor: ${err.message}`, err, err.stack);
-        throw new DBOSInitializationError(err.message);
-      } else {
-        throw err;
-      }
+      this.logger.error(new Error(`failed to initialize workflow executor: ${(err as Error).message}`));
+      throw new DBOSInitializationError(`failed to initialize workflow executor: ${(err as Error).message}`);
```

Comment:
These constructs lose the stack - is that OK?

Reply:
The logger will have the stack: either it creates its own Error instance, either it captures the one from the input Error object. I n this case, the latter

Reply:
I don't follow what you are saying.  The old code gave err.stack to the logger.  In both cases a new error is made which loses the original stack... but the new code makes no record of it.  This is the kind of thing that can make our code hard for even ourselves to debug - if you want that detail you have to rebuild the SDK with a change, then reinstall the package.

Reply:
I see, yes you are absolutely right, the new stack will not include any frame that went "deeper" in the `try` statement above